### PR TITLE
--fix chat

### PR DESCRIPTION
**Pull request by Pervuy**

### CODE REVIEW

File: src/main/java/org/tinder/dao/MessageDao.java [hunk 1]
```diff
@@ -74,7 +74,15 @@ private String getTextQueryMessages(){
 
     @Override
     public boolean save(Message message) throws Exception {
-        throw new RuntimeException("Not implement");
+        PreparedStatement preparedStatement = conn.prepareStatement("""
```
Comment (BohdanRadchenko) [inline src/main/java/org/tinder/dao/MessageDao.java:77]: Could you please use SQlRequester class?

File: src/main/java/org/tinder/servlets/MessageServlet.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    }
+
+    // http://localhost:8080/messages
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+
+        Integer id = getMessageId(req);
+        String token = CookieWorker.getCookieOrThrow(req, CookieNames.AUTH_TOKEN);
+        User user = new UserService().login(token);
```
Comment (BohdanRadchenko) [inline src/main/java/org/tinder/servlets/MessageServlet.java]: we already has user service in services. 

you can use MessageServlet extends ServicesServlet
and super services

services prop ->  new MessageServlet(services) in TinderApp

File: src/main/java/org/tinder/servlets/MessageServlet.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.List;
+
+public class MessageServlet extends HttpServlet{
+
+    private final MessageService messageService;
+
+    public MessageServlet(){
```
Comment (BohdanRadchenko) [inline src/main/java/org/tinder/servlets/MessageServlet.java]: MessageServlet extends ServicesServlet

File: src/main/resources/tmp/chat.ftl [hunk 4]
```diff
@@ -0,0 +1,90 @@
+<#import "common/template.ftl" as t>
+<#import "common/links.ftl" as links>
+<#import "common/script.ftl" as script>
+
+<@t.page>
+    <@t.head title="Chat">
+        <@links.fac/>
+    </@t.head>
+
+    <body>
```
Comment (BohdanRadchenko) [inline src/main/resources/tmp/chat.ftl:10]: use body from template

<@t.body> 

</@t.body>

File: src/main/java/org/tinder/servlets/MessageServlet.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        } catch (TemplateException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        Integer id = getMessageId(req);
+        String token = CookieWorker.getCookieOrThrow(req, CookieNames.AUTH_TOKEN);
+        User user = new UserService().login(token);
```
Comment (BohdanRadchenko) [inline src/main/java/org/tinder/servlets/MessageServlet.java]: use services.user.login(token)

File: src/main/java/org/tinder/servlets/MessageServlet.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        String message = req.getParameter("message");
+        messageService.saveMessage(message, user,id);
+        resp.sendRedirect(String.format("/messages/%s", id));
+    }
+
+    private Integer getMessageId(HttpServletRequest req){
```
Comment (BohdanRadchenko) [inline src/main/java/org/tinder/servlets/MessageServlet.java:65]: imo,this is the responsibility of the filter

File: src/main/java/org/tinder/servlets/MessageServlet.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+                    .getTemplate("chat.ftl")
+                    .process(dataModel, printWriter);
+        } catch (TemplateException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        Integer id = getMessageId(req);
```
Comment (BohdanRadchenko) [inline src/main/java/org/tinder/servlets/MessageServlet.java:57]: u can create filter and use setAttribute for set chat id

for example
in filter
req.setAttribute("id", 0)

in servlet
req.getAttribute("id")

return 0 ?

File: src/main/java/org/tinder/dao/MessageDao.java [hunk 8]
```diff
@@ -25,7 +22,10 @@ public ResultSet getMessagesByChatId(Integer chatId) throws SQLException {
         PreparedStatement preparedStatement =
                                 conn.prepareStatement(
```
Comment (BohdanRadchenko) [inline src/main/java/org/tinder/dao/MessageDao.java:23]: we have SQLRequester

File: src/main/java/org/tinder/filters/MessagesStaticResourcesFilter.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+import org.tinder.services.Services;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class MessagesStaticResourcesFilter extends RequestFilter{
```
Comment (BohdanRadchenko) [inline src/main/java/org/tinder/filters/MessagesStaticResourcesFilter.java]: can u create this filter for all path in app (/*)

File: src/main/java/org/tinder/filters/MessagesStaticResourcesFilter.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    @Override
+    boolean accept(HttpServletRequest req, HttpServletResponse res) {
+
+        String path = req.getRequestURI().substring(req.getContextPath().length());
+        return !path.startsWith("/messages/static/");
+    }
+
+    @Override
+    void failed(HttpServletRequest req, HttpServletResponse res) {
+        String newPath = req.getRequestURI()
```
Comment (BohdanRadchenko) [inline src/main/java/org/tinder/filters/MessagesStaticResourcesFilter.java]: this is redundant code

just request dispatcher from req.pathInfo

path /messages/static/css/style.css 
req.getRequestURI() =  /messages/static/css/style.css 
req.pathInfo() = /static/css/style.css (path without servlet context)

File: src/main/java/org/tinder/filters/MessagesStaticResourcesFilter.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class MessagesStaticResourcesFilter extends RequestFilter{
+
+    public MessagesStaticResourcesFilter(Services services) {
+        super(services);
+    }
+    @Override
+    boolean accept(HttpServletRequest req, HttpServletResponse res) {
```
Comment (BohdanRadchenko) [inline src/main/java/org/tinder/filters/MessagesStaticResourcesFilter.java]: just return ! req.pathInfo start from /static

File: src/main/java/org/tinder/services/MessageService.java [hunk 12]
```diff
@@ -15,25 +15,44 @@
...
+    public MessageService() {
 
         this.messageDao = new MessageDao(Database.getConnection());
     }
 
-    public List<Message> getMessagesByChatId(Integer chat_id) throws SQLException {
+
+    public boolean saveMessage(String content, User user, Integer chatID){
+        if (content == null) return false;
+        if (content.length() == 0) return false;
```
Comment (BohdanRadchenko) [inline src/main/java/org/tinder/services/MessageService.java:26]: if (content == null || content.length() == 0) return false

File: src/main/java/org/tinder/servlets/MessageServlet.java (hunk 2 above)
Comment (Pervuy) [reply src/main/java/org/tinder/servlets/MessageServlet.java]: ok

File: src/main/java/org/tinder/servlets/MessageServlet.java (hunk 3 above)
Comment (Pervuy) [reply src/main/java/org/tinder/servlets/MessageServlet.java]: ok

File: src/main/java/org/tinder/servlets/MessageServlet.java (hunk 5 above)
Comment (Pervuy) [reply src/main/java/org/tinder/servlets/MessageServlet.java]: ok

File: src/main/java/org/tinder/filters/MessagesStaticResourcesFilter.java (hunk 11 above)
Comment (Pervuy) [reply src/main/java/org/tinder/filters/MessagesStaticResourcesFilter.java]: ok

File: src/main/java/org/tinder/filters/MessagesStaticResourcesFilter.java (hunk 10 above)
Comment (Pervuy) [reply src/main/java/org/tinder/filters/MessagesStaticResourcesFilter.java]: ok

File: src/main/java/org/tinder/filters/MessagesStaticResourcesFilter.java (hunk 9 above)
Comment (Pervuy) [reply src/main/java/org/tinder/filters/MessagesStaticResourcesFilter.java]: ok

**Pull request merged**